libs/cpp-ledger/filecoin/impl/ledger_filecoin_impl.cpp

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "cpp-ledger/filecoin/impl/ledger_filecoin_impl.hpp"
+
+#include "cpp-ledger/filecoin/impl/utils.hpp"
+
+namespace ledger::filecoin {
+
+  const VersionInfo requiredVersion{
+      .appMode = 0, .major = 0, .minor = 3, .patch = 0};
+
+  constexpr Byte kCLA = 0x06;
+
+  constexpr Byte kINSGetVersion = 0;
+  constexpr Byte kINSGetAddrSECP256K1 = 1;
+  constexpr Byte kINSSignSECP256K1 = 2;
+
+  constexpr Byte kPayloadChunkInit = 0;
+  constexpr Byte kPayloadChunkAdd = 1;
+  constexpr Byte kPayloadChunkLast = 2;
+
+  constexpr size_t kHardenCount = 2;
+
+  constexpr size_t kMinResponseLength = 4;
+  constexpr size_t kMinSignLength = 66;
+
+  constexpr size_t kPublicKeyLength = 65;
+
+  LedgerFilecoinImpl::LedgerFilecoinImpl(
+      const std::shared_ptr<LedgerDevice> &ledgerDevice)
+      : device(ledgerDevice) {}
```

Comment:
```suggestion
      std::shared_ptr<LedgerDevice> ledgerDevice)
      : device(std::move(ledgerDevice)) {}
```

Reply:
For what?

Reply:
https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference/51706303

Reply:
In general, move semantic is not always better. And I'm not sure it makes sense for shared_ptr, but ok I change it